tests/test_pacman_mirrors.py

```diff
@@ -16,137 +16,81 @@
 class TestPacmanMirrors(unittest.TestCase):
     """Pacman Mirrors Test suite"""
     def setUp(self):
+        """Setup tests"""
         pass
 
     @patch("os.getuid")
     def test_run(self, mock_os_getuid):
         """Run pacman-mirrors"""
         mock_os_getuid.return_value = 0
+
         with unittest.mock.patch("sys.argv",
                                  ["pacman-mirrors",
-                                  "-g",
-                                  "-m", "random",
-                                  "-d", "data/mirrors"]):
+                                  "--quiet",
+                                  "-m", "random"]):
             app = pacman_mirrors.PacmanMirrors()
-            app.config_file = "conf/pacman-mirrors.conf"
-            app.config = app.config_init()
+            app.config = app.load_conf()
             app.command_line_parse()
-            app.load_server_lists()
+            app.load_mirror_file()
+            app.validate_custom_config()
+            app.validate_country_selection()
+            app.gen_server_lists()
+            if app.interactive:
+                app.gen_mirror_list_interactive()
+            else:
+                app.gen_mirror_list_common()
 
     @patch("os.getuid")
-    def test_run_country(self, mock_os_getuid):
-        """Single country via commandline"""
+    def test_country_commandline(self, mock_os_getuid):
+        """Single country from argument"""
         mock_os_getuid.return_value = 0
         with unittest.mock.patch("sys.argv",
                                  ["pacman-mirrors",
-                                  "-g",
-                                  "-c", "Germany",
-                                  "-m", "random",
-                                  "-d", "data/mirrors"]):
+                                  "-c", "Germany"]):
```

Comment:
I am working on splitting internals down to manageable testable units..
The actual server_list generation runs in a separate function which is omitted in everything else but the test run of the entire app.

Test entire app
```
with unittest.mock.patch("sys.argv",
                         ["pacman-mirrors",
                          "--quiet",
                          "-m", "random"]):
    app = pacman_mirrors.PacmanMirrors()
    app.config = app.load_conf()
    app.command_line_parse()
    app.load_mirror_file()
    app.validate_custom_config()
    app.validate_country_selection()    
    app.gen_server_lists()
        if app.interactive:
            app.gen_mirror_list_interactive()
        else:
            app.gen_mirror_list_common()
```
other don't include the lines ranking/randomizing and mirrorlist creation
```
app.gen_server_lists()
if app.interactive:
    app.gen_mirror_list_interactive()
else:
    app.gen_mirror_list_common()
```